src/unlink/unlink.rs

```diff
@@ -69,9 +70,10 @@ pub fn uumain(args: Vec<String>) -> i32 {
 
     let st_mode = {
         let mut buf: stat = unsafe { uninitialized() };
+        let c_string = CString::new(matches.free[0].clone()).unwrap(); // unwrap() cannot fail, the string comes from argv so it cannot contain a \0.
```

Comment:
Doesn't this NUL-terminated string need to be used with `unlink()` as well?

Reply:
Indeed. Fixed.

Reply:
I guess that was the cause of the failure earlier?